website/content/blog/2023-11-reduce-reuse-rebase-buildpacks.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "Reduce, Reuse, Rebase: Sustainable Containers with Buildpacks"
+description: Buildpacks are the greenest choice for container builds
+slug: 2023-11-reduce-reuse-rebase-buildpacks
+weight: 1
+date:   2023-11-17 00:00:00 +0000
+author: Joe Kutner
+---
+
+Container builds can be very wasteful. Every operating system update, new dependency version, and toolchain upgrade results in copious amounts of energy used to build and rebuild our container images; often unnecessarily. It can be costly at scale, which is why Cloud Native Buildpacks were designed to perform builds as efficiently as possible.
+
+Buildpacks are the greenest choice for container builds because the rebaseable layers and advanced caching mechanisms they provide can prevent many images from being rebuilt at all. That's a big shift from many cloud native technologies that assume unlimited resources are available in the cloud.
+
+<p class="mt-5 mb-5"><img src="/images/blogs/2023-11-reduce-reuse-rebase-buildpacks/cnb-green-logo.png" alt="CNB logo with recycling"></p>
+
+## On the Environmental Impact of Cloud Native
+
+Prior to the emergence of the cloud native ecosystem and widespread use of container images, we deployed our applications in servers built from machine images that were updated infrequently and on a different cadence from the application itself.
+
+Today, many applications are coupled to the operating system and its packages because they use a Dockerfile to define their container images. As a result, those images frequently need rebuilds to apply patches to OS-level components, or simply to update tools that aren't even used by the application. Even worse, the layer caching mechanism imposed by Dockerfile forces us to frequently rebuild layers that don't even need to be rebuilt.
+
+The cloud-native ecosystem has brought great productivity and operational improvements to software development. But we've lost sight of how wasteful some of those technologies can be.
```

Comment:
> But we've lost sight of how wasteful some of those technologies can be.
@jkutner Can you elaborate what do you mean by wasteful here?

Reply:
What do you think about something like this?

> The cloud-native ecosystem has brought great productivity and operational improvements to software development. But we've lost sight of how wasteful some of those technologies can be. The relative ease and low cost of cloud infrastructure encourages us to solve problems by spinning up new resources instead of optimizing for the efficient use of our resources. We see this with container builds using Dockerfile, but also other parts of CI/CD, where it's sometimes easier to re-run expensive processes.